tooling/src/hypothesistooling/__main__.py

```diff
@@ -492,6 +492,16 @@ def check_rust_tests():
     cr.cargo("test")
 
 
+@rust_task
+def format_conjecture_rust_code():
+    cr.cargo("fmt")
+
+
+@rust_task
+def check_conjecture_rust_format():
+    cr.cargo("fmt", "--", "--check")
+
+
```

Comment:
I'm wondering if we should instead have a single `lint`, `check-format`, and `format` job which runs all the formatters if-and-only-if there are code changes in the respective directories.  Still fast in the common case, but easy to remember.

Reply:
Another option is to combine these ideas and have combined jobs + sub-jobs for each sub-project. But if there is no much time difference in running everything combined then we can go with combined jobs as you mention. In practice, I usually run `cargo fmt` on file save via IDE integration, since it is fast enough to be not noticeable  :) But for CI it makes complete sense to have a job that checks all formatting.

However, there are already tasks for Ruby lints in CI - I went the same way and added a format job to the same place as other linting & formatting jobs are.

Reply:
I.e I think we can rearrange lint & format tasks in a separate PR